tests/python/gaia-ui-tests/gaiatest/apps/settings/regions/device_info.py

```diff
@@ -42,28 +62,73 @@ def model(self):
     def software(self):
         return self.marionette.find_element(*self._software_locator).text
 
+    def tap_your_rights(self):
+        element = self.marionette.find_element(*self._your_rights_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._rights_page_locator))
+
+    def tap_your_privacy(self):
+        element = self.marionette.find_element(*self._your_privacy_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._privacy_page_header_locator))
+
+    def tap_legal_info(self):
+        element = self.marionette.find_element(*self._legal_info_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        Wait(self.marionette).until(expected.element_displayed(*self._open_source_notices_locator))
+
+    def tap_open_source_notices(self):
+        element = self.marionette.find_element(*self._open_source_notices_locator)
+        Wait(self.marionette).until(expected.element_displayed(element))
+        element.tap()
+        iframe = self.marionette.find_element(*self._notice_text_locator)
+        Wait(self.marionette).until(expected.element_displayed(iframe))
+        self.marionette.switch_to_frame(iframe)
+        _last_child_locator = (By.CSS_SELECTOR, 'body > *:last-child')
```

Comment:
I didn't think of this, but :last-child is dynamic, so while the page is loading, :last-child is just something that keeps on getting updated to whatever the latest element is during page load.
I guess it's not a problem here, because the page is more or less loaded instantly, but I guess ideally you need an id or unique selector here.